blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorManager.java

```diff
@@ -56,26 +57,31 @@ public void onPause() {
      *
      * @param workspace the workspace to request code generation for.
      * @param codeGenerationCallback the callback to call with the generated code.
-     * @throws BlocklySerializerException if the workspace cannot be serialized to XML.
      */
     public void requestCodeGeneration(List<String> blockDefinitionsJsonPaths,
                                       List<String> generatorsJsPaths,
                                       Workspace workspace,
-                                      CodeGeneratorCallback codeGenerationCallback)
-        throws BlocklySerializerException {
+                                      CodeGeneratorCallback codeGenerationCallback) {
 
         if (workspace.hasBlocks()) {
             if (isBound()) {
-                final StringOutputStream serialized = new StringOutputStream();
-                workspace.serializeToXml(serialized);
-
-                mGeneratorService.requestCodeGeneration(
-                    new CodeGenerationRequest(
-                        serialized.toString(),
-                        codeGenerationCallback,
-                        blockDefinitionsJsonPaths,
-                        generatorsJsPaths)
-                );
+                try {
+                    final StringOutputStream serialized = new StringOutputStream();
+                    workspace.serializeToXml(serialized);
+
+                    mGeneratorService.requestCodeGeneration(
+                        new CodeGenerationRequest(
+                            serialized.toString(),
+                            codeGenerationCallback,
+                            blockDefinitionsJsonPaths,
+                            generatorsJsPaths)
+                    );
+                } catch (BlocklySerializerException e) {
```

Comment:
This was in the AbstractActivity to allow developers to override the error handling. If you're interested in making it a bit cleaner I'd suggest adding an onError method to CodeGeneratorCallback with a default implementation of showing a toast then calling it from here.